ykrt/src/compile/jitc_yk/opt/simple.rs

```diff
@@ -34,24 +34,50 @@ fn opt_mul(
     rhs: PackedOperand,
 ) -> Result<(), CompilationError> {
     match (lhs.unpack(m), rhs.unpack(m)) {
-        (Operand::Local(mul_inst), Operand::Const(mul_const))
-        | (Operand::Const(mul_const), Operand::Local(mul_inst)) => {
+        (Operand::Local(mut mul_inst), Operand::Const(mul_const))
+        | (Operand::Const(mul_const), Operand::Local(mut mul_inst)) => {
             let old_const = m.const_(mul_const);
-            if let Some(y) = old_const.int_to_u64() {
-                if y == 0 {
+            if let Some(old_val) = old_const.int_to_u64() {
+                let mut new_val = old_val;
+                // If we've got `%2: mul %1, xi8` then see if `%1` is of the form `mul %0, yi8`: if so
+                // we've got a chain that's `%2: %0 x*y`. We can thus "skip" the intermediate `mul`
```

Comment:
I think the chain should be `%2: %0*y*x`.

Reply:
`*` is commutative, so the two are equivalent.

Reply:
Fine, but you're still missing a `*`.

Reply:
Fixed in d668e29917b0dfc697329338a83520193b1e82fa.